framework_tvb/tvb/core/services/project_service.py

```diff
@@ -790,7 +790,7 @@ def _review_operation_inputs(self, operation_gid):
         parameters = json.loads(operation.parameters)
         try:
             adapter = ABCAdapter.build_adapter(operation.algorithm)
-            return adapter.review_operation_inputs(parameters)
+            return adapter.review_operation_inputs(operation)
```

Comment:
I think we should also change the exception handling here. Remove TODO once this is done.